TS.fsx

```diff
@@ -531,8 +531,25 @@ let EmitDictionaries flavor =
         | "Object" -> Pt.printl "interface %s {" dict.Name
         | _ -> Pt.printl "interface %s extends %s {" dict.Name dict.Extends
 
+        let emitJsonProperty (p: ItemsType.Root) =
+            Pt.printl "%s: %s;" p.Name.Value p.Type.Value
+
+        let removedPropNames = 
+            getRemovedItems ItemKind.Property flavor 
+            |> Array.filter (matchInterface dict.Name)
+            |> Array.map (fun rp -> rp.Name.Value)
+        let addedProps = 
+            getAddedItems ItemKind.Property flavor 
+            |> Array.filter (matchInterface dict.Name)
+
         Pt.increaseIndent()
-        dict.Members |> Array.iter (fun m -> Pt.printl "%s?: %s;" m.Name (DomTypeToTsType m.Type))
+        Array.iter emitJsonProperty addedProps
+        dict.Members
+        |> Array.filter (fun m -> not (Array.contains m.Name removedPropNames))
+        |> Array.iter (fun m -> 
+            match (findOverriddenItem m.Name ItemKind.Property dict.Name) with
+            | Some om -> emitJsonProperty om
+            | None -> Pt.printl "%s?: %s;" m.Name (DomTypeToTsType m.Type))
```

Comment:
It is more or less subjective when it comes to the syntax preferences though. I used the continuation style because it makes the data flow clearer, and avoids the burden of naming things. Though I agree both styles have merits.